### PR TITLE
Adding Build Environment

### PR DESCRIPTION
**Pull request by VenkyOS**
Signed-off-by: Venkataraman Padmanabhan<venkataraman.padmanabhan@seagate.com>

### CODE REVIEW

File: doc/Build_Environment.rst [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+- Functions, which we know are not standard or may be different on some platforms (direct I/O);
+
+- Optional things, such as debug on/off or things like whether to build or not server or client, should be moved to build/configuring options in a way like this: debug=1;
+
+- Any software packages we need to build the project, such as db4 or Linux kernel, should be specified in the way like this: with-gss=gss-path with-linux=linux-kernel-path.
+
+The following configuring options form configure part functionality:
+
+.. image:: Images/Few.PNG
```
Comment (huanghua78) [inline doc/Build_Environment.rst:90]: No need to use an image here.
We can create a table here to have those contents.

Text can be easily modified, but not an image.
If we really want to use image, please consider use PLANTUML.

Comment (VenkyOS) [reply doc/Build_Environment.rst]: Incorporated.

File: doc/Build_Environment.rst [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+===============
+
+Generation part is very simplistic for all of the build systems described. One just specifies what targets needed: binaries, libraries, etc. The only exception is that, we may possibly need handling for not standard source files, such as fol template, to generate fol tables for db4.
+
+Repository Path
+================
+
+So far there are three repositories for storing different logical parts of the Motr source tree:
+
+.. image:: Images/REP.PNG
```
Comment (huanghua78) [inline doc/Build_Environment.rst:155]: Remove this image.
Use table, please.

The same as the above one.

Comment (VenkyOS) [reply doc/Build_Environment.rst]: Incorporated

File: doc/Build_Environment.rst [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+|                        | Repository for storing motr-core source code.                                             |
++------------------------+-------------------------------------------------------------------------------------------+
+|Motr-db4                | Git clone ssh:/gitosis@git.clusterstor.com/motr-db4.git                                   |
++------------------------+-------------------------------------------------------------------------------------------+
+
+As one can see, motr-core and motr-db4 are separate repositories. There is no some logical explanation why we need them separately now as there are no requirements dictating this structure. We surely can have such a requirements in future and more flexible repository structure allows to adopt quickly. Additionally, there are not many drawbacks in current structure to not do so.
+
+While motr-db4 repository is having completely the same structure as standard db4 distribution (we do not want to multiply differences, so that we could switch to new version quickly), motr-core, on the other hands, having the structure, discussed below.
```
Comment (huanghua78) [inline doc/Build_Environment.rst:178]: Now, please update this table with our CURRENT git configurations.
and remove "motr-db4" related sentences and paragraphes.


```
cortx-motr: https://github.com/Seagate/cortx-motr
cortx-motr-galois: https://github.com/Seagate/cortx-motr-galois
```

**Review by mukul-seagate11 — APPROVED** [review]
looks good

**Comment by huanghua78** [comment]
The Few.PNG and REP.PNG are not used.

**Comment by VenkyOS** [comment]
> The Few.PNG and REP.PNG are not used.

Incorporated.

**Review by huanghua78 — APPROVED** [review]

**Pull request merged**